develop/2017-05-29-fgrehm-analysis-on-meals-based-on-presence-information.py

```diff
@@ -0,0 +1,467 @@
+
+# Reduce dataset to meals with matching company and state info
+meals = pd.merge(
+    meals, 
+    companies[['cnpj', 'city', 'state', 'main_activity', 'name']], 
+    left_on='cnpj_cpf', 
+    right_on='cnpj',
+    suffixes=('_congressperson', '_company')
+)
+print("Meals with matching companies:", len(meals))
+
+meals = meals[~meals.state_company.isnull()]
+print("Meals with known company state:", len(meals))
+meals = meals.rename(index=str, columns={
+    "state_congressperson": "congressperson_state",
+    "name": "company_name", 
+    "city": "company_city", 
+    "state_company": "company_state"
+})
+
+
+# In[9]:
+
+texts = pd.read_csv('../data/2017-02-15-receipts-texts.xz', dtype={'text': np.str}, low_memory=False)
+texts['text'] = texts.text.str.upper()
+
+meals_with_ts = texts.merge(meals[['document_id']], on='document_id')
+print("Meals with OCR:", len(meals_with_ts))
+
+
+# In[10]:
+
+def extract_timestamps(text):
+    return re.findall('[0-9][0-9]:[0-9][0-9]:[0-9][0-9]', str(text))
+meals_with_ts['timestamps'] = meals_with_ts.text.apply(extract_timestamps)
+
+def parse_timestamps(ts):
+    try:
+        return time.strptime(ts, "%H:%M:%S")
+    except:
+        return None
+meals_with_ts['timestamps'] = meals_with_ts.timestamps.apply(lambda ts: list(map(parse_timestamps, ts)))
+meals_with_ts = meals_with_ts[meals_with_ts.apply(lambda r: len(r.timestamps) > 0, axis=1)]
+
+meals = pd.merge(
+    left=meals,
+    right=meals_with_ts[['document_id', 'timestamps']],
+    how='left',
+    on='document_id'
+)
+print("Meals with timestamps:", len(meals[~meals.timestamps.isnull()]))
+
+
+# In[11]:
+
+meals = meals[[
+    'document_id',
+    'applicant_id',
+    'congressperson_document',
+    'congressperson_id',
+    'congressperson_name',
+    'congressperson_state',
+    'party',
+    'year',
+    'issue_date',
+    'total_net_value',
+    'company_name',
+    'company_state',
+    'timestamps'
+]]
+
+
+# ## Meals outside DF while the congressperson was present in a session
+# 
+# There are at least two ways we can identify that:
+# * Making an approximation of the total time the congressperson was in Brasilia (like more than X hours) and checking for reimbursements on days they were there for a long period of time
+# * Leveraging OCR data from receipts and grabbing their timestamps
+
+# In[12]:
+
+deputies = pd.read_csv('../data/2017-04-19-deputies.xz', low_memory=False)
+print("Total deputies:", len(deputies))
+
+sessions = pd.read_csv('../data/2017-04-19-session-start-times.xz', low_memory=False)
+sessions['date'] = pd.to_datetime(sessions['date'], format="%Y-%m-%dT%H:%M:%S").dt.date
+sessions['started_at'] = pd.to_datetime(sessions['started_at'], format="%Y-%m-%dT%H:%M:%S")
+print("Session records:", len(sessions))
+
+presences = pd.read_csv('../data/2017-04-19-presences.xz', low_memory=False)
+presences['date'] = pd.to_datetime(presences['date'], format="%Y-%m-%dT%H:%M:%S").dt.date
+presences.sort_values('date', ascending=False)
```

Comment:
`sort_values` will in fact generate a new sorted dataset, not sort `presences` in places. Since in the [line 194](https://github.com/datasciencebr/serenata-de-amor/pull/247/files#diff-d54592a5a813f2ab6c4164a0849cb127R194) you perform an aggregation using the `date` field, you probably want to add `inplace=True` there.